torch/csrc/autograd/engine.cpp

```diff
@@ -72,23 +73,42 @@ struct GraphTask {
   // Notified when a task finishes executing.  Check outstanding_tasks to see
   // if all tasks are done.
   std::condition_variable not_done;
-  const Engine::pre_callback_map& pre_callbacks;
-  const Engine::post_callback_map& post_callbacks;
   std::unordered_map<Function*, InputBuffer> not_ready;
   std::unordered_map<Function*, int> dependencies;
 
+  struct ExecInfo {
```

Comment:
What is an ExecInfo? What is a Capture?

Reply:
It's something that records extra metadata about execution of Functions - e.g. if any of their inputs should be captured (i.e. `input_idx`th input should be saved into `output_idx`th output in the captures vector).